psi4/src/psi4/libmints/zora.h

```diff
@@ -0,0 +1,314 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License along
+ * with Psi4; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * @END LICENSE
+ */
+
+#ifndef _psi_src_lib_libmints_zora_h_
+#define _psi_src_lib_libmints_zora_h_
+
+#include "psi4/libmints/typedefs.h"
+#include <map>
+
+namespace psi {
+
+class BasisSet;
+class DFTGrid;
+class Options;
+class BasisFunctions;
+
+/*! \ingroup MINTS
+ *  \class ZORA
+ *  \brief Scalar relativistic kinetic integral through zeroth-order regular approximation (ZORA).
+ *
+ *  \details Here we use the model potential proposed by van Wüllen in 
+ *  *Molecular density functional calculations in the regular relativistic approximation*
+ *  (https://doi.org/10.1063/1.476576/). The exact values of the model potential come from
+ *  NWCHEM (https://www.nwchem-sw.org/). Given atomic orbital :math:`\chi`, we evaluate the
+ *  following integral on a grid:
+ *
+ *  :math:`T^\text{SR}_{ij}=\int d\mathbf{r}^3 \frac{c^2}{2c^2-v_\text{eff} (\mathbf{r})}
+ *  \nabla\chi_i^\dagger(\mathbf{r}) \cdot\nabla\chi_j(\mathbf{r})`
+ *
+ *  The effective potential :math:`v_\text{eff}(\mathbf{r})` is the potential experienced
+ *  at a point due to nuclear attraction and electron repulsion. The generated matrix
+ *  replaces the kinetic energy for SCF etc. Because this is only evaluated once, don't
+ *  cheap out on the grid! The ZORA time is usually negligible.
+ *
+ *  \author Nathan Gillispie, University of Memphis, 2025
+ */
+class PSI_API ZORA {
+   public:
+	ZORA(std::shared_ptr<Molecule>, std::shared_ptr<BasisSet>, Options&);
+	~ZORA();
+
+    /*! @{
+     * Fills an empty SharedMatrix with the computed ZORA kinetic integral.
+     * @param T Shared matrix object that will hold the ZORA kinetic integrals.
+     */
+    void compute(SharedMatrix);
+    /*! @} */
+
+   private:
+    std::shared_ptr<Molecule> molecule_;
+    std::shared_ptr<BasisSet> primary_;
+	std::shared_ptr<DFTGrid> grid_;
+	Options& options_;
+    /*! Purpose of the map: the ``block->index()`` does not index like ``grid_->blocks()``.
+     *  For multithreading to work, this is a solution.
+     */
+    std::shared_ptr<std::map<int, SharedVector>> veff_;
+
+
+	/// Initializes grid_ with options, prints details
+	void setup();
+	/// Fills veff_ with effective potential for each point in a block.
+	void compute_veff();
+	/// Creates same structure as above but all zeros. TSR->T as veff->0
+	void compute_debug_veff();
+	/// Uses veff_ to compute the Scalar Relativistic Kinetic Energy Matrix
+	void compute_TSR(std::vector<std::shared_ptr<BasisFunctions>>, SharedMatrix&);
+
+    /*! I'm so sorry.
+     *  These are the contraction coefficients and exponents of the model basis. The
+     *  last array indexes the atoms by atomic number.
+     */
+	static constexpr double coeffs[2919] = {0.0063990067,0.0745660339,0.3008381442,0.4565180734,0.1616787418,
```

Comment:
Nothing more to address here, I just find it absolutely hilarious whenever a header file is longer than the corresponding `.cc` file 🤣